create_coda_files/create_coda_files.py

```diff
@@ -1,55 +1,153 @@
+        help='Path to coding scheme that is for this data')
+    parser.add_argument(
+        'is_yes_no', metavar='is-yes-no',
+        help='Is this variable a yes no question')
+    parser.add_argument(
+        'auto_cleaner', metavar='auto-cleaner',
+        help='automated cleaner to apply to this data')
 
     args = parser.parse_args()
     user = args.user
     json_input_path = args.json_input_path
     flow_name = args.flow_name
     variable_name = args.variable_name
+    traced_json_output_path = args.traced_json_output_path
     coda_output_path = args.coda_output_path
-    if not os.path.exists(prev_coda_path):
-        prev_coda_path = None
+    coding_scheme_path = args.coding_scheme_path
+    is_yes_no = args.is_yes_no
+    if args.auto_cleaner in AUTO_CLEANERS.keys():
+        auto_cleaner = args.auto_cleaner
+        auto_cleaner = AUTO_CLEANERS[auto_cleaner]
     else:
-        prev_coda_path = args.prev_coda_path
+        print('There is no matching cleaner for: {}'.format(args.auto_cleaner))
+        auto_cleaner = None
 
-    # Load data from JSON file
-    with open(json_input_path, "r") as f:
-        show_messages = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+    # Load in the coding scheme
+    code_scheme = _open_scheme(coding_scheme_path)
+
+    # Load traced data list from JSON file
+    with open(json_input_path, 'r') as f:
+        list_td = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
 
     # Filter out test messages sent by AVF.
-    show_messages = [td for td in show_messages if not td.get("test_run", False)]
+    list_td = [td for td in list_td 
+                if not td.get('test_run', False)]
+
+    # Keys used in the script
+    message_key = '{} (Text) - {}'.format(variable_name, flow_name)
+    coded_key = '{}_coded'.format(variable_name.lower())
+    id_field = '{}_id'.format(message_key)
+    time_key = '{} (Time) - {}'.format(variable_name, flow_name)
+    if is_yes_no == 'True':
+        is_yes_no = True
+        yes_no_key = '{}_yesno'.format(variable_name.lower())
+        yes_no_scheme = _open_scheme('Yes_No.json')
+        yes_no_cleaner = AUTO_CLEANERS['yes_no']
+    else:
+        is_yes_no = False
+        yes_no_cleaner = None
+
+    # Label data missing the key as true_missing
+    # Data with empty strings will also be labelled true missing
+    # create list that will be output to coda
+    coda_td = []
+    for td in list_td:
+        if td.get(message_key, '') == '':
+            missing_dict = dict()
+            na_label = CleaningUtils.make_cleaner_label(
+                        code_scheme,
+                        code_scheme.get_code_with_control_code(
+                            Codes.TRUE_MISSING),
+                        Metadata.get_call_location()
+                    )
+            missing_dict[coded_key] = na_label.to_dict()
+            if is_yes_no:
+                missing_dict[yes_no_key] = na_label.to_dict()
```

Comment:
Isn't your yes/no scheme different? In which case you need to assign the na_label for your yes/no scheme, not for the other scheme